projects/miopen/src/solver/tensorOp/Op5dTensorGenericContiguous.cpp

```diff
@@ -0,0 +1,190 @@
+#include <miopen/tensor.hpp>
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/float_equal.hpp>
+#include <miopen/datatype.hpp>
+
+namespace miopen {
+
+namespace solver {
+
+namespace tensorOp {
+
+bool Op5dTensorGenericContiguous::IsApplicable(
+    [[maybe_unused]] const ExecutionContext&,
+    const miopen::tensorOp::ProblemDescription& problem) const
+{
+    const auto& aTensorDesc = problem.GetATensorDesc();
+    const auto& bTensorDesc = problem.GetBTensorDesc();
+    const auto& cTensorDesc = problem.GetCTensorDesc();
+
+    if(aTensorDesc.GetLengths().size() != 5)
+        return false;
+    if(aTensorDesc.GetLengths() != bTensorDesc.GetLengths())
+        return false;
+    if(bTensorDesc.GetLengths() != cTensorDesc.GetLengths())
+        return false;
+
+    if(!(aTensorDesc.IsContiguous() && bTensorDesc.IsContiguous() && cTensorDesc.IsContiguous()))
+        return false;
+
+    if(!(aTensorDesc.AllLengthsFitIntoInt() && bTensorDesc.AllLengthsFitIntoInt() &&
+         cTensorDesc.AllLengthsFitIntoInt()))
+        return false;
+
+    const auto& cl = cTensorDesc.GetLengths();
+    uint64_t total = 1;
+    for(int i = 0; i < 5; ++i)
+    {
+        const uint64_t t = static_cast<uint64_t>(cl[i]);
+        if(t == 0)
+            return false;
+        if(total > (std::numeric_limits<uint64_t>::max() / t))
+            return false;
+        total *= t;
+    }
+    if(total > std::numeric_limits<uint32_t>::max())
+        return false;
+
+    return true;
+}
+
+std::size_t Op5dTensorGenericContiguous::GetWorkspaceSize(
+    [[maybe_unused]] const ExecutionContext&,
+    [[maybe_unused]] const miopen::tensorOp::ProblemDescription&) const
+{
+    return 0;
+}
+
+ConvSolution
+Op5dTensorGenericContiguous::GetSolution(const ExecutionContext& context,
+                                         const miopen::tensorOp::ProblemDescription& problem) const
+{
+    ConvSolution result{miopenStatusSuccess};
+
+    const auto& cTensorDesc = problem.GetCTensorDesc();
+    const auto& clens       = cTensorDesc.GetLengths();
+
+    KernelBuildParameters build_params;
+    GetCommonParams(build_params, problem, false);
+    build_params.Define("USE_5D_TENSOR_GENERIC_CONTIGUOUS");
+
+    // Use 32-bit verified all dimensions and total elements fit
+    build_params.Define("USE_INDEX32");
+
+    // PACK_T - how many consecutive elements each thread handles per inner iteration.
+    // declaration + define as main source, in kernel defined as fallback - only
+    constexpr size_t k_PACK_T = 8;
+    build_params.Define("PACK_T", k_PACK_T);
+
+    const uint64_t total_elems_u64 =
+        static_cast<uint64_t>(clens[0]) * static_cast<uint64_t>(clens[1]) *
+        static_cast<uint64_t>(clens[2]) * static_cast<uint64_t>(clens[3]) *
+        static_cast<uint64_t>(clens[4]);
+
+    const size_t total_elems   = static_cast<size_t>(total_elems_u64);
+    const size_t total_packets = (total_elems + k_PACK_T - 1) / k_PACK_T;
+
```

Comment:
Please a reference to generic kernel heuristic caps here, or paste the whole comment here
```suggestion
/*!
* For more context @ref cu_scaled_wg_cap "CU‑scaled WG cap heuristic".
*/
```